django/forms/models.py

```diff
@@ -669,7 +670,10 @@ def save_m2m():
                 for form in self.saved_forms:
                     form.save_m2m()
             self.save_m2m = save_m2m
-        return self.save_existing_objects(commit) + self.save_new_objects(commit)
+        if not self.edit_only:
+            return self.save_existing_objects(commit) + self.save_new_objects(commit)
+        else:
+            return self.save_existing_objects(commit)
```

Comment:
I think we are duplicating functionality with handling `edit_only` attribute. The save behaviour can already be achieved by adding this to the Inline class:
```  python
    def has_add_permission(self, request, obj):
        return False
```
Assuming I'm not missing something, please let me know.

If overriding this method gives the desired effect, then perhaps we can only check for this new `edit_only` attribute in the default implementation of `has_add_permission` method?

Also, does `edit_only` preclude deleting existing inline instances? If yes, we can do the same check in `has_delete_permission` method.

Reply:
Or perhaps follow what has been implemented with `can_delete` attribute. It is as an "and" with `has_delete_permission`: https://github.com/django/django/search?p=1&q=can_delete

Maybe adding a new `can_add` attribute in a similar way would be more consistent with the current code?